webui/react/src/hooks/useModal/Experiment/useModalExperimentMove.tsx

```diff
@@ -65,21 +65,26 @@ const useModalExperimentMove = ({ onClose }: Props): ModalHooks => {
   const [sourceProjectId, setSourceProjectId] = useState<number | undefined>();
   const [experimentIds, setExperimentIds] = useState<number[]>();
   const { canMoveExperimentsTo } = usePermissions();
-  const workspaces = Loadable.map(useWorkspaces({ archived: false }), (ws) =>
+  const loadableWorspaces = useWorkspaces({ archived: false });
+  const workspaces = Loadable.map(loadableWorspaces, (ws) =>
     ws.filter((w) => canMoveExperimentsTo({ destination: { id: w.id } })),
   );
   const projects = useWorkspaceProjects(workspaceId);
   const ensureProjectsFetched = useEnsureWorkspaceProjectsFetched(canceler.current);
-  const ensureWorkspacesFetched = useEnsureWorkspacesFetched(canceler.current);
+  const fetchWorkspaces = useFetchWorkspaces(canceler.current);
 
   const handleClose = useCallback(() => onClose?.(), [onClose]);
 
   const { modalOpen: openOrUpdate, modalRef, ...modalHook } = useModal({ onClose: handleClose });
 
   useEffect(() => {
-    ensureWorkspacesFetched();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    const isLoadingWorkspaces = Loadable.isLoading(loadableWorspaces);
+    const loaded = Loadable.isLoaded(loadableWorspaces);
+
+    if (isLoadingWorkspaces) return;
```

Comment:
These two conditions are exactly the same so the fetch will never fire. You should create a `ensureWorkspacesFetched` function for this usage.